src/include/cmark-gfm_config.h

```diff
@@ -1,30 +1,24 @@
+#if __has_include("config.h")
```

Comment:
This pattern doesn't make sense to me - can you please add extensive comments on each instance for why this is needed.  I'm concerned that it can result in recursive includes.

Reply:
The idea behind this dance is to use the CMake-generated config.h if it's available - that's guaranteed to be applicable to the current compilation environment, as opposed to the guesses that are in the prebuilt headers used by the Swift package. The CMake-generated headers use the same header guard as the prebuilt ones - hence putting this block outside the header guard here. I've pushed a commit that tweaks how these headers are detected (as part of scrubbing out `__has_include`), and provides a comment explaining a little of what's going on.